src/api/sci_api.cc

```diff
@@ -36,11 +36,162 @@
 #include "sparse_ci/sparse_ci_solver.h"
 #include "integrals/active_space_integrals.h"
 
+#include "sparse_ci/determinant.h"
+#include "sparse_ci/determinant_hashvector.h"
+#include "sparse_ci/sparse_state_vector.h"
+#include "sparse_ci/sparse_operator.h"
+#include "sparse_ci/sparse_fact_exp.h"
+#include "sparse_ci/sparse_exp.h"
+#include "sparse_ci/sparse_hamiltonian.h"
+
 namespace py = pybind11;
 using namespace pybind11::literals;
 
 namespace forte {
 
+/// Export the Determinant class
+void export_Determinant(py::module& m) {
+    py::class_<Determinant>(m, "Determinant")
+        .def(py::init<>())
+        .def(py::init<const Determinant&>())
+        .def(py::init<const std::vector<bool>&, const std::vector<bool>&>())
+        .def("get_alfa_bits", &Determinant::get_alfa_bits, "Get alpha bits")
+        .def("get_beta_bits", &Determinant::get_beta_bits, "Get beta bits")
+        .def("nbits", &Determinant::get_nbits)
+        .def("norb", &Determinant::norb)
+        .def("get_alfa_bit", &Determinant::get_alfa_bit, "n"_a, "Get the value of an alpha bit")
+        .def("get_beta_bit", &Determinant::get_beta_bit, "n"_a, "Get the value of a beta bit")
+        .def("set_alfa_bit", &Determinant::set_alfa_bit, "n"_a, "value"_a,
+             "Set the value of an alpha bit")
+        .def("set_beta_bit", &Determinant::set_beta_bit, "n"_a, "value"_a,
+             "Set the value of an beta bit")
+        .def("create_alfa_bit", &Determinant::create_alfa_bit, "n"_a, "Create an alpha bit")
+        .def("create_beta_bit", &Determinant::create_beta_bit, "n"_a, "Create a beta bit")
+        .def("destroy_alfa_bit", &Determinant::destroy_alfa_bit, "n"_a, "Destroy an alpha bit")
+        .def("destroy_beta_bit", &Determinant::destroy_beta_bit, "n"_a, "Destroy a beta bit")
+        .def("count_alfa", &Determinant::count_alfa, "Count the number of set alpha bits")
+        .def("count_beta", &Determinant::count_beta, "Count the number of set beta bits")
+        .def(
+            "gen_excitation",
+            [](Determinant& d, const std::vector<int>& aann, const std::vector<int>& acre,
+               const std::vector<int>& bann,
+               const std::vector<int>& bcre) { return gen_excitation(d, aann, acre, bann, bcre); },
```

Comment:
Where is `gen_excitation` defined? Same question applies `str` below.

Reply:
Added a comment (`// uses str() defined in determinant.hpp`)